### ISSUE TITLE
`detach_` behaves differently when computing the gradients in forward mode w/ `jit`

### ISSUE THREAD

**Issue by payphon** [pytorch/pytorch#85568]
### 🐛 Describe the bug

`detach_` behaves differently when computing the gradients in forward mode w/ `jit`

```py
import torch
def m(input):
    return input.detach_()

device = 'cuda'

input = torch.empty([1], dtype=torch.float32, device=device)

jit_m = torch.jit.trace(m, (input.clone(), ))

from torch.autograd.functional import jacobian

try:
    print(jacobian(m, input.clone().requires_grad_(), vectorize=True, strategy='forward-mode'))
except Exception as e:
    print(e)

print(jacobian(jit_m, input.clone().requires_grad_(), vectorize=True, strategy='forward-mode'))
```

```
Can't detach views in-place. Use detach() instead. If you are using DistributedDataParallel (DDP) for training, and gradient_as_bucket_view is set as True, gradients are views of DDP buckets, and hence detach_() cannot be called on these gradients. To fix this error, please refer to the Optimizer.zero_grad() function in torch/optim/optimizer.py as the solution.
tensor([[0.]], device='cuda:0')
```

Interestingly, the reverse mode can give a gradient without jit

```py
import torch
def m(input):
    return input.detach_()

device = 'cuda'

input = torch.empty([1], dtype=torch.float32, device=device)

jit_m = torch.jit.trace(m, (input.clone(), ))

from torch.autograd.functional import jacobian
print(jacobian(m, input.clone().requires_grad_(), vectorize=True, strategy='reverse-mode'))
# tensor([[0.]], device='cuda:0')
```

I am not sure which way is correct, but they do need to behave consistently.

### Versions

```
Collecting environment information...
PyTorch version: 1.13.0.dev20220923+cu116
Is debug build: False
CUDA used to build PyTorch: 11.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04.1 LTS (x86_64)
GCC version: (Ubuntu 11.2.0-19ubuntu1) 11.2.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

Python version: 3.9.13 (main, Aug 25 2022, 23:26:10)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.19.5-051905-generic-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3060
Nvidia driver version: 515.65.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.3
[pip3] torch==1.13.0.dev20220923+cu116
[pip3] torchaudio==0.13.0.dev20220923+cu116
[pip3] torchvision==0.14.0.dev20220923+cu116
[conda] numpy                     1.23.3                   pypi_0    pypi
[conda] torch                     1.13.0.dev20220923+cu116          pypi_0    pypi
[conda] torchaudio                0.13.0.dev20220923+cu116          pypi_0    pypi
[conda] torchvision               0.14.0.dev20220923+cu116          pypi_0    pypi

```

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
